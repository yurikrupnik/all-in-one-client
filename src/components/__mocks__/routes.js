/* eslint react/jsx-filename-extension: 0 */

// import React from 'react';
import Shows from '../Shows';
import MobxMovies from '../MobxMovies';

const routes = [
    {
        path: '/',
        component: Shows,
        key: 'shows'
    },
    {
        path: '/mobx',
        component: MobxMovies,
        key: 'mobx'
    }
];

export default routes;
