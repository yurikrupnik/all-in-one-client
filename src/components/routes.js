
import Shows from './Shows';
import ContextMovies from './ContextMovies';
import MobxMovies from './MobxMovies';
import AngularMovies from './AngularMovies';
import VueMovies from './VueMovies';
import ReduxMovies from './ReduxMovies';

const routes = [
    {
        path: '/shows',
        component: Shows,
        key: 'Shows'
    },
    {
        path: '/context',
        component: ContextMovies,
        key: 'ContextMovies'
    },
    {
        path: '/redux',
        component: ReduxMovies,
        key: 'redux'
    },
    {
        path: '/mobx',
        component: MobxMovies,
        key: 'MobxMovies'
    },
    {
        path: '/vue',
        component: VueMovies,
        key: 'vue'
    },
    {
        path: '/angular',
        component: AngularMovies,
        key: 'angular'
    },
];

export default routes;
