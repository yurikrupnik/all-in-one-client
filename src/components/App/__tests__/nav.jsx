import { cleanup } from 'react-testing-library';
import React from 'react';
import MainNav from '../nav';
import routes from '../../routes';

jest.mock('../../routes'); // eslint-disable-line no-undef

const {
    test,
    expect,
    shallow,
    afterEach
} = global;

afterEach(cleanup);

test('renders <Nav /> component', () => {
    const wrapper = shallow(<MainNav routes={routes} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').length).toBe(2);
});
