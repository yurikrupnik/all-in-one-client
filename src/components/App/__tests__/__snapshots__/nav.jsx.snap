// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Nav /> component 1`] = `
<header>
  <div
    key="shows"
  >
    <Link
      replace={false}
      to="/"
    >
      shows
    </Link>
  </div>
  <div
    key="mobx"
  >
    <Link
      replace={false}
      to="/mobx"
    >
      mobx
    </Link>
  </div>
</header>
`;
